File: Lua/fizzbuzz3.lua
LIMIT = 100

t = {
	[0] = "FizzBuzz",
	[3] = "Fizz", [6] = "Fizz", [9] = "Fizz", [12] = "Fizz",
	[5] = "Buzz", [10] = "Buzz"
	}

for i = 1, LIMIT do
	io.write(t[i % 15] or i)
	if i < LIMIT then
		io.write(", ")
	end
end

print()

File: Pascal/fizzbuzz3.pas
program FizzBuzz3;
const
	limit = 100;
type
	tA = array[0..14] of string;
var
	i: integer;
	s: string;
	a: tA = ('FizzBuzz', '', '', 'Fizz', '', 'Buzz', 'Fizz',
	         '', '', 'Fizz', 'Buzz', '', 'Fizz', '', '');
begin
	for i := 1 to limit do
	begin
		s := a[i mod 15];
		if s = '' then write(i) else write(s);
		if i < limit then write(', ')
	end;
	writeln()
end.
